### PR TITLE
Added DNF configuration and Multimedia Support for fedora and other Redhat Distros

### LINKED ISSUE #289: Dnf Setup
**Is your feature request related to a problem? Please describe.**
Every time install Fedora, I put this into `/etc/dnf/dnf.conf`.
```
defaultyes=True
```
This makes it so that if you press enter on "Install packages? [y/N]", it'll default to yes. It makes it more like apt.

**Describe the solution you'd like**
Add an option to make dnf default yes.

**Describe alternatives you've considered**
Setup manually.

**Additional context**
Seems like someone is working on it in #258

### PR DESCRIPTION
**Pull request by Angxddeep**
# Pull Request

## DNF Configuration & Multimedia Codecs
Just some essential DNF changes to at least make it usable and faster by a margin.
Also added installation support for multimedia codecs

## Type of Change
-  New feature

## Testing
Checked on my System, it works

## Issue related to PR
- Resolves #289

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: tabs/system-setup/fedora/configure-dnf.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+. ../../common-script.sh
+
+configureDNF() {
+    case $PACKAGER in
+        dnf)
+            $ESCALATION_TOOL sed -i '/^max_parallel_downloads=/c\max_parallel_downloads=10' /etc/dnf/dnf.conf || echo 'max_parallel_downloads=10' >> /etc/dnf/dnf.conf
+            echo "fastestmirror=True" | $ESCALATION_TOOL tee -a /etc/dnf/dnf.conf > /dev/null
+            echo "defaultyes=True" | $ESCALATION_TOOL tee -a /etc/dnf/dnf.conf > /dev/null
+
```
Comment (nnyyxxxx) [inline tabs/system-setup/fedora/configure-dnf.sh:11]:
```suggestion

```

File: tabs/system-setup/fedora/configure-dnf.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    case $PACKAGER in
+        dnf)
+            $ESCALATION_TOOL sed -i '/^max_parallel_downloads=/c\max_parallel_downloads=10' /etc/dnf/dnf.conf || echo 'max_parallel_downloads=10' >> /etc/dnf/dnf.conf
+            echo "fastestmirror=True" | $ESCALATION_TOOL tee -a /etc/dnf/dnf.conf > /dev/null
+            echo "defaultyes=True" | $ESCALATION_TOOL tee -a /etc/dnf/dnf.conf > /dev/null
+            
+            $ESCALATION_TOOL "$PACKAGER" -y install dnf-plugins-core
+            ;;
+        *)
+            echo "Unsupported distribution: $DTYPE"
```
Comment (nnyyxxxx) [inline tabs/system-setup/fedora/configure-dnf.sh:15]:
```suggestion
            printf "%b\n" "${RED}Unsupported distribution: $DTYPE${RC}"
```

File: tabs/system-setup/fedora/multimedia-codecs.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+multimedia() {
+    case $PACKAGER in
+        dnf)
+            if [[ -e /etc/yum.repos.d/rpmfusion-free.repo && -e /etc/yum.repos.d/rpmfusion-nonfree.repo ]]; then
```
Comment (nnyyxxxx) [inline tabs/system-setup/fedora/multimedia-codecs.sh:8]: Bashism spotted.

**Review by adamperkowski — APPROVED** [review]
We could do a `multimedia-codecs.sh` in `tabs/system-setup` or `tabs/utils` that would support other distros aswell.

**Comment by adamperkowski** [comment]
> Most distros enable that on the setup

We want this tool to work on **any distro**, **not most**.
Arch Linux (a very popular distro), for example, doesn't bundle ffmpeg in `base`.

**Comment by nnyyxxxx** [comment]
> > Most distros enable that on the setup
> 
> We want this tool to work on **any distro**, **not most**. Arch Linux (a very popular distro), for example, doesn't bundle ffmpeg in `base`.

I agree with Adam.

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**